torchacc/utils/consolidate_and_reshard_ckpts.py

```diff
@@ -0,0 +1,73 @@
+from argparse import ArgumentParser
+from torchacc.dist.state_dict_utils import consolidate_and_reshard_model_dict, consolidate_and_reshard_optim_dict
+
+MODEL_NAME_PATTERN = "rank*-of-*-model.pth"
+OPTIM_NAME_PATTERN = "rank*-of-*-optimizer.pth"
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--ckpt_dir",
+        type=str,
+        required=True,
+        help=(
+            f"The name pattern of the XLA FSDP checkpoint files to be consolidated. "
+            f"Files matching the pattern ``ckpt_dir + ckpt_name`` will be loaded."
+            f"For model, the default pattern is {MODEL_NAME_PATTERN}. For optimizer,"
+            f"the default pattern is {OPTIM_NAME_PATTERN}"),
+    )
+    parser.add_argument(
+        "--ckpt_name",
+        type=str,
+        default="",
+        help=(
+            f"The name pattern of the XLA FSDP checkpoint files to be consolidated. "
+            f"Files matching the pattern ``ckpt_dir + ckpt_name`` will be loaded."
+            f"For model, the default pattern is {MODEL_NAME_PATTERN}. For optimizer,"
+            f"the default pattern is {OPTIM_NAME_PATTERN}"),
+    )
+    parser.add_argument(
+        "--ckpt_type",
```

Comment:
Could we support consolidating the model and optimizer into one command?

Reply:
Users need to tell us the checkpoint file name they want to load and  consolidate. Like "rank-*of-*-model.pth" or "rank-*of-*-optimizer.pth" .What's more, they need to tell us the dir to save.  We now support users to tell us one name_pattern at once. So they need to consolidate model and optimizer seperately.
We can add args like ``model_ckpt_name``, ``optimizer_ckpt_name``, ``model_save_name``, ``optimizer_save_name`` in the script to support consolidating model and optimizer into one command. 
However, I think may be there are some scenarios that users only want to consolidate model or optimizer?